web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java

```diff
@@ -224,7 +224,10 @@ public void setGtfsLinkMappings() {
                                                    HTreeMap<String, String> gtfsLinkMappings) {
         List<String> rowsForFeed = Lists.newArrayList();
         for (String routeId : routeIdToTripsInRoute.keySet()) {
-            for (String tripIdInRoute : routeIdToTripsInRoute.get(routeId)) {
+	    if (!routeIdToTripsInRoute.containsKey(routeId)) {
```

Comment:
How is this even possible? `routeId` comes from the key set of `routeIdToTripsInRoute`, so `routeIdToTripsInRoute.containsKey(routeId)` has to be true, right?

Reply:
From [here](https://replicahq.slack.com/archives/CK358RL0Z/p1614849937013400?thread_ts=1600553031.022500&cid=CK358RL0Z), it looks like the original error appeared on this line: 

```for (Pair<Stop, Stop> stopStopPair : tripIdToStopPairsInTrip.get(tripIdInRoute)) {```

So, I think we want to do a check if `tripIdToStopPairsInTrip.containsKey(tripIdInRoute)`

Reply:
For some additional context:

The origin of `routeIdToTripsInRoute` comes from taking every route ID from `routes.txt`, and mapping them to trips that appear in `trips.txt` and have a `route_id` that lines up with the ID from `routes.txt`. 

`tripIdToStopPairsInTrip` comes from looking in `stop_times.txt`, making a mapping where the keys are the trip IDs appearing in the `trip_id` column for each "stop time" and the values are the set of "stop times" that have that trip ID, and then ordering each list of stop times (ie, the values in this intermediate map we've made) by departure time and grabbing their lat/lons to get a final mapping of (trip ID) --> (set of sequentially-ordered stop-stop pairs) (this is what `tripIdToStopPairsInTrip` holds). We use those sequentially-ordered stop-stop pairs to then do the actual routing between each set of stops (for each trip) later on.

So, I think the cause of the bug must be that there's a trip in trips.txt that doesn't appear in the `trip_id` column of any stop time in `stop_times.txt`. Not well-formatted gtfs, but `continue` definitely seems like the right behavior here.

Reply:
You’re right mark. When i was originally debugging the err i fixed/tested this on a different branch that ultimately had too many merge conflicts. i attempted to remake the change from memory but put it on the wrong map 😅  Just tested again now.

Can’t wait for those tests ;)